pandas/core/reshape/reshape.py

```diff
@@ -291,7 +291,12 @@ def get_new_values(self, values, fill_value=None):
         # if our mask is all True, then we can use our existing dtype
         if mask_all:
             dtype = values.dtype
-            new_values = np.empty(result_shape, dtype=dtype)
+            if isinstance(dtype, ExtensionDtype):
```

Comment:
Could you combine this branch with the one below?  They are very similar

Reply:
I'm not sure it's way better, but I've tried.